File: Examples/android/CMeSdkJava/lib/src/main/java/me/connect/sdk/java/LogFileObserver.java
package me.connect.sdk.java;

import android.os.FileObserver;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class LogFileObserver extends FileObserver {
    public String absolutePath;
    public int MAX_ALLOWED_FILE_BYTES;
    public Process process;

    public LogFileObserver(String path, int MAX_ALLOWED_FILE_BYTES) {
        super(path, FileObserver.ALL_EVENTS);
        this.absolutePath = path;
        this.MAX_ALLOWED_FILE_BYTES = MAX_ALLOWED_FILE_BYTES;
        //this.startLogcat(this.absolutePath);
    }

    @Override
    public void onEvent(int event, String path) {

        //data was written to a file
        if ((FileObserver.MODIFY & event) != 0) {
            File logFile = new File(this.absolutePath);
            if(logFile.length() > MAX_ALLOWED_FILE_BYTES) {
                try {
                    new FileWriter(this.absolutePath).close();
                } catch(IOException ex) {
                    ex.printStackTrace();
                }
            }
        } else if((FileObserver.DELETE_SELF & event) != 0) {
            this.stopWatching();
            try {
                new FileWriter(this.absolutePath).close();
            } catch(IOException ex) {
                ex.printStackTrace();
            }
            this.startWatching();
        }
    }

    // public void stopLogcat() {
    //     this.process.destroy();
    // }

    // public void startLogcat(String filename) {
    //     try {
    //         String cmd = "logcat -f "+filename;
    //         this.process = Runtime.getRuntime().exec(cmd);
    //     } catch (IOException e) {
    //         e.printStackTrace();
    //     }
    // }
}

File: examples/android/MSDKSampleAppKotlin/app/src/main/java/msdk/kotlin/sample/logger/LogFileObserver.kt
package msdk.kotlin.sample.logger

import android.os.FileObserver
import java.io.File
import java.io.FileWriter
import java.io.IOException

class LogFileObserver//this.startLogcat(this.absolutePath); //     this.process.destroy();
// }
// public void startLogcat(String filename) {
//     try {
//         String cmd = "logcat -f "+filename;
//         this.process = Runtime.getRuntime().exec(cmd);
//     } catch (IOException e) {
//         e.printStackTrace();
//     }
// }
    (var absolutePath: String, var MAX_ALLOWED_FILE_BYTES: Int) :
    FileObserver(absolutePath, ALL_EVENTS) {
    var process: Process? = null
    override fun onEvent(event: Int, path: String?) {

        //data was written to a file
        if (MODIFY and event != 0) {
            val logFile = File(absolutePath)
            if (logFile.length() > MAX_ALLOWED_FILE_BYTES) {
                try {
                    FileWriter(absolutePath).close()
                } catch (ex: IOException) {
                    ex.printStackTrace()
                }
            }
        } else if (DELETE_SELF and event != 0) {
            stopWatching()
            try {
                FileWriter(absolutePath).close()
            } catch (ex: IOException) {
                ex.printStackTrace()
            }
            startWatching()
        }
    } // public void stopLogcat() {

}